magento/magento2
# Fix "Required parameter 'theme_dir' was not passed" error when emulating an environment inside an emulated area

### Description
Environment emulation inside an area emulation returns the error "Required parameter 'theme_dir' was not passed". The classes `\Magento\Theme\Model\View\Design` and `\Magento\Theme\Model\Theme` retrieve their area from the app state unless the area is emulated (see their `getArea()` method). When the environment is emulated inside an area emulation, the app state's area is used. This causes incorrect behaviour when the environment is being emulated.

### Manual testing scenarios

```php
use \Magento\Framework\App\Area;

class Emulation
{
    /**
     * @var \Magento\Store\Model\App\Emulation
     */
    protected $appEmulation;

    /**
     * @var \Magento\Framework\App\State
     */
    protected $appState;

    public function __construct(
        \Magento\Store\Model\App\Emulation $appEmulation,
        \Magento\Framework\App\State $appState
    ) {
        $this->appEmulation = $appEmulation;
        $this->appState = $appState;
    }

    public function execute()
    {
        $this->appState->emulateAreaCode(Area::AREA_ADMINHTML, function () {
            $this->appEmulation->startEnvironmentEmulation(1, Area::AREA_FRONTEND, true);
            $this->appEmulation->stopEnvironmentEmulation();
        });
    }
}
```

I've executed this class using a cronjob (the cache needs to be flushed before starting). I have a store with the id "1" (lets call this store "default"). The default store uses the theme `Magento/luma`. `startEnvironmentEmulation(..)` at some point tries to load the default store's theme. `\Magento\Theme\Model\Theme getFullPath()` gets invoked and returns `adminhtml/Magento/luma`. The theme's full path is wrong as the app state's area is being used instead of the environments area, because the app state is emulated. This then prevents the theme from being loaded and produces an error. I've added a method to also check if the environment is being emulated and used it to load the correct area. 

### Contribution checklist
